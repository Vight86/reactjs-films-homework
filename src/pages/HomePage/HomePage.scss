.main {
  padding-right: max(calc((100% - 2120px) / 2), 2em);
  padding-left: max(calc((100% - 2120px) / 2), 2em);
}
