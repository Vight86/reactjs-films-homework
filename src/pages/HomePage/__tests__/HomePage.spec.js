import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import HomePage from '../index';

const setUp = () => {
  const renderer = new ShallowRenderer();
  renderer.render(<HomePage />);
  return renderer.getRenderOutput();
};

describe('render HomePage component', () => {
  it('should be main tag', () => {
    const result = setUp();
    expect(result.type).toBe('main');
  });

  it('render component without props', () => {
    const result = setUp();
    expect(result).toMatchSnapshot();
  });
});
