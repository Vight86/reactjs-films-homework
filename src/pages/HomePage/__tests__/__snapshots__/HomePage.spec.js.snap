// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render HomePage component render component without props 1`] = `
<main
  className="main"
>
  <Header
    className="secondary"
  >
    <FilterList>
      <Button
        className="filterButton"
        onClick={[Function]}
      >
        Trending
      </Button>
      <Button
        className="filterButton"
        onClick={[Function]}
      >
        Top rated
      </Button>
      <Button
        className="filterButton"
        onClick={[Function]}
      >
        Coming soon
      </Button>
      <SelectButton
        genres={
          Array [
            "action",
            "scifi",
          ]
        }
      />
    </FilterList>
    <ToggleButton
      className="toggleGridButton"
    />
  </Header>
  <MovieList
    movies={
      Array [
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
        Object {
          "genre": Array [
            "scifi",
            "comedy",
          ],
          "image": Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          },
          "info": "There are growing dangers in
    the wizarding world of 1926 New York.
    Something mysterious is leaving a path of destruction in the streets,
    threatening to expose the wizarding",
          "rating": "3.9",
          "title": "Independance day",
        },
      ]
    }
  />
</main>
`;
